### PR TITLE
add `--debug` flag to `package-docker` to compile in debug mode

### PR DESCRIPTION
**Pull request by terencehonles**

### CODE REVIEW

**Review by ashkulz — CHANGES_REQUESTED** [review]
@terencehonles why do you need this? I'd be completely opposed to implementing this as-is, because there's no sense in building a `debug` package. I looked into figuring out how to build debug .deb/.rpm packages, but that wasn't supported in fpm at the time.

At the very least, the filename must change so that it's clear that it's a debug package.

**Comment by terencehonles** [comment]
Well it seems useful to be able to pass the flag through. Otherwise you need to know how to use the `compile-docker` command (which expects different arguments) and then you have to figure out how to install that differently than you'd normally install a package. I would be fine with it altering the name that it's using, but I'm not completely sure where I might put that.

Unfortunately I'm running into an issue where a page is segfaulting when I'm trying to generate a page that requested with an accept-language header "es". It works in both "en" and "fr" which I'm a little confused by (I expected it to break for "fr" too). I built a debug package with `--debug` as patched and unfortunately it looks like by just enabling the debug mode whatever it was segfaulting on is not an issue.

I looked and found `-force-debug-info`, but apparently that's only for Qt5. I currently just finished building with `-separate-debug-info` but I _don't_ think that actually will leave whatever optimizations are likely causing the segfault, but any tips would be appreciated. I would ideally like to build `-release`, but still have the debug symbols _somewhere_.

File: build [hunk 1]
```diff
@@ -131,12 +131,12 @@ def compile_docker(config, target, src_dir, tgt_dir, debug=False):
...
-def package_docker(config, target, src_dir, iteration, clean=False):
+def package_docker(config, target, src_dir, iteration, clean=False, debug=False):
     tgt_dir = os.path.join('targets', target)
     if clean:
         shell('rm -fr %s' % tgt_dir)
 
-    compile_docker(config, target, src_dir, tgt_dir)
+    compile_docker(config, target, src_dir, tgt_dir, debug=debug)
 
     pkg_ver, pkg_iter = get_version(src_dir, iteration)
```
Comment (ashkulz) [inline build:141]:
```suggestion
    pkg_ver, pkg_iter = get_version(src_dir, iteration)
    if debug:
        pkg_iter += ".debug"
```
Can you check if this is used, it generates a different package name? In that case, I might be okay to merge it with this implemented.

Comment (terencehonles) [reply build:141]: Looking at the code that looks like this probably should work, testing a build right now.

Do you know if there's a reason `--clean` both deletes the target dir before and after compiling? I wanted a clean build from the last build I ran, but was expecting to look at the intermediate files and didn't realize it nuked the directory after the build. I ran it with the wrong arguments so I needed to re-run anyways, but can I remove the `rm -fr` post build and just leave the pre-build clean up?

Comment (terencehonles) [reply build:141]: This built `wkhtmltox_0.12.7-0.20220705.dev.debug.buster_amd64.deb` as expected.

File: build [hunk 2]
```diff
@@ -131,14 +131,18 @@ def compile_docker(config, target, src_dir, tgt_dir, debug=False):
...
+    if debug:
+        pkg_iter += ".debug"
```
Comment (ashkulz) [inline build:143]:
```suggestion
```

**Review by ashkulz — APPROVED** [review]

**Comment by ashkulz** [comment]
Thanks for the contribution!

**Comment by terencehonles** [comment]
@ashkulz I'm still experimenting on how to build a release w/ debug symbols, any thoughts or advice?

I believe may have just built with the same optimization level by patching the Qt source qmake file and will test that shortly. I would think distributing an additional package something like https://packages.debian.org/sid/wkhtmltopdf-dbgsym might be something that can be emulated.

**Pull request merged**